packages/react/src/r4b/provider.tsx

```diff
@@ -1,33 +1,26 @@
 import { Formatter } from "@bonfhir/core/r4b";
-import { PropsWithChildren, createElement, useMemo } from "react";
+import { PropsWithChildren, createElement } from "react";
 import { FhirUIContext, OnNavigateArgs } from "./context";
 import { FhirUIDefaultProps } from "./default-props";
+import { useFhirFormatters } from "./formatters/fhir-formatters-context";
+import { FhirFormattersProvider } from "./formatters/provider";
 import { FhirUIRenderer } from "./renderer";
 
-export type FhirUIProviderProps = PropsWithChildren<{
-  formatter?: Formatter | null | undefined;
+// DO NOT EXPORT, this is to support the independent formatters context
+type DeprecationWrapperProps = PropsWithChildren<{
   renderer: Partial<FhirUIRenderer>;
   defaultProps?: FhirUIDefaultProps | null | undefined;
   onNavigate?: (args: OnNavigateArgs) => void;
 }>;
 
-/**
- * Provide a context for ui components.
- * Should probably be placed near the top of your React app render tree.
- */
-export function FhirUIProvider(props: FhirUIProviderProps) {
-  const formatter = useMemo(() => {
-    if (props.formatter) {
-      return props.formatter;
-    }
-
-    return Formatter.default;
-  }, [props.formatter]);
+// DO NOT EXPORT, this is to support the independent formatters context
+const FhirUIProviderWrapper: React.FC<DeprecationWrapperProps> = (props) => {
+  const { formatter } = useFhirFormatters();
 
   return (
     <FhirUIContext.Provider
       value={{
-        formatter,
+        formatter: formatter,
```

Comment:
Nitpick: This can use shorthand notation.

Reply:
done!